### PR TITLE
Add tracking analytics (Tracks) to the Backup plugin.

### PR DESCRIPTION
**Pull request by elliottprogrammer**
This PR adds Tracks tracking to the Backup plugin, and adds tracking events to the links in the Backup dashboard UI.

#### Changes proposed in this Pull Request:
- Adds `jetpack-analytics' dependency to package.json.
- Added a new `useAnalytics()` hook to initialize analytics for the Backup plugin.
- Added a tracking event for the Backup plugin admin **page-view**. (`jetpack_backup_admin_page_view`)
- Added a tracking event for the "**See backups in the cloud**" CTA button click. (`jetpack_backup_see_backups_cta_click`)
- Added a tracking event for the "**Or view your most recent restore point**" link click, (`jetpack_backup_view_recent_restore_points_click`)
- Added a tracking event for the "**Learn more**" link click. (`jetpack_backup_learn_more_click`)
- Added a tracking event for the "**See all your backups**" link click. (`jetpack_backup_see_all_backups_click`)
- Added a tracking event for the "**See your site's activity**" link click. (`jetpack_backup_see_site_activity_click`)

Screenshot:
** Red arrows indicate the links where tracking was added.
Markup 2022-07-07 at 16 58 52

#### Other information:

- [X] Have you written new tests for your changes, if applicable? **N/A**
- [X] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

Project Thread: p1HpG7-gb4-p2

Tracking and design changes are further described in this i2 design P2: p1HpG7-gkG-p2

This PR is a continuation of: https://github.com/Automattic/jetpack/pull/24955

#### Does this pull request change what data or activity we track or use?

Yes. It adds some link click tracking.

#### Testing instructions:

- Spin up a JN site with Jetpack Beta, using Backup branch `add/backup-plugin-analytics`.
- Connect Backup and purchase Backup plan with credits (or add Backup(10GB) product using Store Admin).
- Go to https://cloud.jetpack.com/settings/:site?action=edit (replace :site with your JN site slug) and **enter your JN site server credentials**.
- Navigate to the Jetpack --> Backup page (`/wp-admin/admin.php?page=jetpack-backup`).
- Open developer tools console and enter: `localStorage.setItem('debug', 'dops:analytics');`
- Reload/refresh the Backup admin page.
- When the page reloads, verify you see the `jetpack_backup_admin_page_view` event in the console.
- Click the "**See backups in the cloud**" CTA button. Verify the `jetpack_backup_see_backups_cta_click` event fires.
- Click the "**Or view your most recent restore point**" link, verify the `jetpack_backup_view_recent_restore_points_click` event fires.
- Click the "**Learn more**" link. Verify the `jetpack_backup_learn_more_click` event fires.
- Click the "**See all your backups**" link. Verify the `jetpack_backup_see_all_backups_click` event fires.
- Click the "**See your site's activity**" link. Verify the `jetpack_backup_see_site_activity_click` event fires.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1202580352248681

### CODE REVIEW

File: projects/packages/backup/src/js/hooks/useAnalytics.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		clearedIdentity,
+		ga,
+		mc,
+		pageView,
+		purchase,
+		setGoogleAnalyticsEnabled,
+		setMcAnalyticsEnabled,
+		setProperties,
+		tracks,
+	};
```
Comment (manzoorwanijk) [inline projects/packages/backup/src/js/hooks/useAnalytics.js]: Any specific reason for destructuring and not directly returning `jetpackAnalytics` here?

**Review by manzoorwanijk — DISMISSED** [review]
Works as expected, just have the above question related to `useAnalytics` hook

File: projects/packages/backup/src/js/hooks/useAnalytics.js (hunk 1 above)
Comment (elliottprogrammer) [reply projects/packages/backup/src/js/hooks/useAnalytics.js]: Hmm, yes good point!  I'm not sure why I destructured these first. Thanks Manzoor!
I'll change this and directly return `jetpackAnalytics` instead.

**Review by coder-karen — DISMISSED** [review]
Tests well (test on JN initially).

An aside (because from what I can see it doesn't come from this PR):
I do however somehow have a broken backup connection within my local development build, which means I have access to the backups page in wp-admin but also see a message saying 'we're having trouble backing up your site'.  

There is a console error: `Warning: useCallback received a final argument that is not an array (instead, received `string`). When specified, the final argument must be an array.`. The line flagged is line 22 of Admin.js: `	const [ connectionStatus, renderConnectScreen ] = useConnection();`.

My understanding of React isn't yet strong enough to identify the issue, however it seems it isn't relevant for this particular PR - approving this one as is.

**Comment by elliottprogrammer** [comment]
> An aside (because from what I can see it doesn't come from this PR):

@coder-karen  thank you for testing and for your review! 🙂

Hmm, yes, I am also able to reproduce the console error, and I agree that it does not appear to be related to this PR.
I'll try to uncover and resolve the error in a following PR. 👍

**Comment by elliottprogrammer** [comment]
Hi @coder-karen 👋, some conflicts appeared so I had to merge-in trunk and run `fix-up-project-versions.sh`, which dismissed your previous approval. Would you mind giving this a re-review?

Also, I see in the checks, there is a "`Release in progress / Release Lead approval required`". Was that there before when you reviewed the first time?  My apologies, this is somewhat new to me and I don't really know what that means, or if it's correct, or if maybe I did something incorrectly? I know _I did_ update package dependencies, (I added `@automattic/jetpack-analytics` ), and maybe that triggers a new release, so maybe thats why?. Well anyway, I just bring it up to make sure I'm not doing something wrong. Thanks so much! Let me know if anything doesn't look quite right to you. 🙂

**Review by coder-karen — APPROVED** [review]
> Would you mind giving this a re-review?

Done!

> Also, I see in the checks, there is a "Release in progress / Release Lead approval required". Was that there before when you reviewed the first time

That would have been because at the time of your comment the release lock was on - it was on whilst another release was being prepared and was on for much longer than usual: p1657641897002629-slack-CDLH4C1UZ

That generally just means that the PR can't be merged to trunk during that period so as to avoid conflicts happening while preparing the release (unless specifically approved for release during the lock), but you should still be able to click the button to enable auto-merge which would merge the PR as soon as the lock is off.

**Comment by elliottprogrammer** [comment]
r248990-wpcom

**Pull request merged**